packages/client/src/runtime/core/model/aggregates/aggregate.ts

```diff
@@ -0,0 +1,82 @@
+import type { Client } from '../../../getPrismaClient'
+import type { ModelAction } from '../applyModel'
+import { aggregateMap } from './utils/aggregateMap'
+
+/**
+ * Transforms the `userArgs` for the `.aggregate` shorthand. It is an API sugar
+ * for not having to do things like: `{select: {_avg: {select: {age: true}}}}`.
+ * The goal here is to desugar it into something that is understood by the QE.
+ * @param userArgs to transform
+ * @returns
+ */
+export function desugarUserArgs(userArgs: object) {
+  const _userArgs = desugarCountInUserArgs(userArgs)
+  const userArgsEntries = Object.entries(_userArgs)
+
+  return userArgsEntries.reduce(
+    (aggregateArgs, [key, value]) => {
+      if (aggregateMap[key] !== undefined) {
+        // if it's an aggregate, we re-wrap into select
+        aggregateArgs['select'][key] = { select: value }
+      } else {
+        aggregateArgs[key] = value // or leave it alone
+      }
+
+      return aggregateArgs
+    },
+    { select: {} } as object,
+  )
+}
+
+/**
+ * Desugar `userArgs` when it contains `{_count: true}`.
+ * @param userArgs the user input
+ * @returns
+ */
+function desugarCountInUserArgs(userArgs: object) {
+  if (typeof userArgs['_count'] === 'boolean') {
+    return { ...userArgs, _count: { _all: userArgs['_count'] } }
+  }
+
+  return userArgs
+}
+
+/**
+ * Creates an unpacker that adds sugar to the basic result of the QE. An
+ * unpacker helps to transform a result before returning it to the user.
+ * @param userArgs the user input
+ * @returns
+ */
+export function createSugarUnpacker(userArgs: object) {
+  return (data: object) => {
+    if (typeof userArgs['_count'] === 'boolean') {
+      if (Array.isArray(data)) {
```

Comment:
Could add a comment of when it's an array here?

Reply:
Will do. This is shared logic for all aggregates, so arrays only happen when you use group by. Probably not worth saving a few lines of code for the confusion, I'll de-condense.